### PR TITLE
[Cosmos DB] az cosmosdb create: add --key-uri to support adding key vault encryption information

### PR DESCRIPTION
**Pull request by shurd**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S166.

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ def load_arguments(self, _):
 
     with self.argument_context('cosmosdb create') as c:
         c.argument('account_name', completer=None)
+        c.argument('key_vault_key_uri', help="The URI of the key vault")
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py]: Can we mark the parameter as preview using `is_preview=True`?

File: src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py [hunk 2]
```diff
@@ -926,3 +926,19 @@ def test_cosmosdb_table_resource_throughput(self, resource_group):
...
+            'kv_name': self.create_random_name(prefix='cli', length=15),
+            'key_name': self.create_random_name(prefix='cli', length=15),
+            'location': "eastus2"
+        })
+
+        self.cmd('az keyvault create --resource-group {rg} -n {kv_name} --enable-soft-delete true --enable-purge-protection true')
+        self.cmd('az keyvault set-policy -n {kv_name} -g {rg} --spn a232010e-820c-4083-83bb-3ace5fc29d0b --key-permissions get unwrapKey wrapKey')
+        self.cmd('az keyvault key create -n {key_name} --kty RSA --size 3072 --vault-name {kv_name}')
+
+        self.cmd('az cosmosdb create -n {acc} -g {rg} --locations regionName={location} failoverPriority=0 --key-vault-key-uri https://{kv_name}.vault.azure.net/keys/{key_name}')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py]: Can we add some checks or assert values for the command?

**Comment by shurd** [comment]
@Juliehzl I made the suggested changes, and I think I fixed the broken CI tests

**Comment by shurd** [comment]
Will this be merged for the next release?

**Comment by Juliehzl** [comment]
> Will this be merged for the next release?

Yes, I can help you merge it. But I have a small suggestion here: can we make the parameter name shorter e.g. `--key-uri`?

**Comment by shurd** [comment]
@Juliehzl it looks like the pipeline failed because of a network error. Could you rerun the CI? I made the name change

**Comment by Juliehzl** [comment]
@shurd can you help confirm if the title is fine to you as release note? If yes, I will help merge it.

**Comment by shurd** [comment]
Yes the title is good for the release notes

**Pull request merged**